libsrc/telestrat/cputc.s

```diff
@@ -4,26 +4,50 @@
 ; void cputc (char c);
 ;
 
-        .export         _cputc, cputdirect
+        .export         _cputc, _cputcxy, cputdirect, display_conio
         .export         CHARCOLOR, OLD_CHARCOLOR, BGCOLOR, OLD_BGCOLOR
 
+        
+        .import         update_adscr
+        .import         popax
+
         .include        "telestrat.inc"
 
 cputdirect:
+_cputcxy:
+    pha                           ; Save C
+    jsr     popax                 ; Get X and Y
+    sta     SCRY                  ; Store Y
+    stx     SCRX                  ; Store X
+    jsr     update_adscr
+    pla        
+
 .proc _cputc
+    cmp     #$0D
+    bne     @not_CR
+    ldy     #$00
+    sty     SCRX 
+    rts
+@not_CR:
+    cmp     #$0A
+    bne     @not_LF
+
+    inc     SCRY
+    jmp     update_adscr
+
+@not_LF:
```

Comment:
`cputdirect:` should be put just below this label.  `jsr cputdirect` should print characters, but not interpret control characters.

Reply:
@greg-king5  ok i have to look to this too

Reply:
@greg-king5 as i managed conio target, control character are not managed : 
        ldy     SCRX
        sta     (ADSCR),y

i never manage \n \r etc 

It means that if you do 'cputc(0x0D); it will display a background attribute.

If you ask to manage cputdirect, it means that i have to manage also \n \r ?

Reply:
You do manage '\n' and '\r'.  You added lines 26 through 38 into "cputc.s"; they handle both of those controls.  `cputdirect:` must be moved to a line below those added lines.

Look at https://github.com/cc65/cc65/blob/master/libsrc/atmos/cputc.s
It shows where `cputdirect:` belongs.